### ISSUE TITLE
Experiments with highly multilingual mix for cooldown

### ISSUE THREAD

**Issue by vitiugin** [OpenEuroLLM/Taskboard#106]
[Multiblimp evaluation done by Sampo](https://mattermost.ufal.mff.cuni.cz/openeurollm/pl/digcii4q7jg8if7uiwg8r9kg7w) demonstrated that our 2B 4TT models are lower in language fluency compared to 2B 100BT monolingual reference models (both are trained on HPLT 2 cleaned).

Based on multi-staged training with uniform multilingual distribution presented by TildeOpen model we'll try to use equal distribution for decay training stage.

> The foundational model training involves 450,000 updates with a constant batch size of 4,718,592 tokens, using a constant learning rate followed by a cooldown phase across 2 trillion tokens. Training consists of three distinct data sampling phases. First, all languages are sampled uniformly to ensure equal representation. Second, languages are sampled according to their natural distribution to ensure that the model sees as much data from languages with larger speaker bases as possible. Finally, we return to uniform sampling across all languages. This three-phase approach ensures TildeOpen LLM develops balanced multilingual capabilities while maintaining strong performance across all target languages, particularly the underrepresented European languages.

We plan to run two experiments:
- use 2B 4TT model trained on 90% of English + code + math / 10% of multilingual data and apply uniform distribution for decay phase (starting from 765000 iteration)
- use 2B 4TT model trained on 50% of English + code + math / 50% of multilingual data and apply 90% English + code + math distribution for decay phase (starting from 765000 iteration)
